### PR TITLE
feat(circleci): include beanstalk deploy in cicd (#700)

### PR DESCRIPTION
**Pull request by svitalii33**
## Current Behavior

There is no beanstalk deployment today.

## Expected Behavior

This will trigger deploying the updates to beanstalk on tag releases.
## Related Issue(s)

Fixes #

### CODE REVIEW

File: .platform/hooks/prebuild/yarn.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+# need to install node first to be able to install yarn (as at prebuild no node is present yet)
+sudo curl --silent --location https://rpm.nodesource.com/setup_14.x | sudo bash -
```
Comment (webberwang) [inline .platform/hooks/prebuild/yarn.sh:4]: could we remove this, and move to circleci ci. we wouldn't want this to trigger locally

File: package.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "codelab",
   "engines": {
-    "node": ">=13",
```
Comment (webberwang) [inline package.json:4]: we'd still want this requirement, could we make it work on aws

**Review by webberwang — CHANGES_REQUESTED** [review]

File: .ebignore [hunk 3]
```diff
@@ -0,0 +1,25 @@
+/node_modules
```
Comment (webberwang) [inline .ebignore:1]: I think better to ignore all, then re-add required folders

File: .circleci/config.yml [hunk 4]
```diff
@@ -305,6 +310,35 @@ jobs:
...
+            sudo apt-get install python3-pip awscli build-essential
+            pip3 install awsebcli --upgrade --user
+      - run:
+          name: Create AWS credentials manually
+          command: |
+            mkdir ~/.aws
+            touch ~/.aws/config
+            chmod 600 ~/.aws/config
+            echo "[profile eb-cli]" > ~/.aws/config
+            echo "aws_access_key_id=$AWS_ACCESS_KEY" >> ~/.aws/config
```
Comment (webberwang) [inline .circleci/config.yml:330]: we would use vault

File: .circleci/config.yml [hunk 5]
```diff
@@ -83,6 +84,10 @@ executors:
     docker:
       - image: joatmon08/circleci-gcp-agent:287.0-1.14.72
     resource_class: small
+  python:
```
Comment (webberwang) [inline .circleci/config.yml:87]: is this being used?

**Review by webberwang — CHANGES_REQUESTED** [review]

File: .platform/hooks/prebuild/yarn.sh (hunk 1 above)
Comment (svitalii33) [reply .platform/hooks/prebuild/yarn.sh:4]: It is required step to run yarn command in beanstalk

Comment (webberwang) [inline .platform/hooks/prebuild/yarn.sh:4]: Can this file be removed now that we have correct node version?

File: .circleci/config.yml [hunk 6]
```diff
@@ -306,6 +306,35 @@ jobs:
...
+      - run:
+          name: Create AWS credentials manually
+          command: |
+            mkdir ~/.aws
+            touch ~/.aws/config
+            chmod 600 ~/.aws/config
+            echo "[profile eb-cli]" > ~/.aws/config
+            echo "aws_access_key_id=$AWS_ACCESS_KEY" >> ~/.aws/config
+            echo "aws_secret_access_key=$AWS_SECRET_KEY" >> ~/.aws/config
+      - run:
```
Comment (webberwang) [inline .circleci/config.yml:328]: We have a job for this right?

**Review by webberwang — CHANGES_REQUESTED** [review]

File: .platform/hooks/prebuild/yarn.sh (hunk 1 above)
Comment (svitalii33) [reply .platform/hooks/prebuild/yarn.sh:4]: As I mentioned in the comment, above command is required to install yarn in prebuild stage.

File: .platform/hooks/prebuild/yarn.sh [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+sudo wget https://dl.yarnpkg.com/rpm/yarn.repo -O /etc/yum.repos.d/yarn.repo
+sudo yum -y install yarn
+
+# install
+cd /var/app/staging/
+
+# debugging..
+ls -lah
+
+yarn install --frozen-lockfile --prod=true
```
Comment (webberwang) [inline .platform/hooks/prebuild/yarn.sh:13]: is this required?

**Review by webberwang — CHANGES_REQUESTED** [review]

File: .circleci/config.yml [hunk 8]
```diff
@@ -306,6 +306,32 @@ jobs:
...
+          name: Deploying to eb
+          command: |
+            eb use app-staging --region ap-east-1 --profile eb-cli
```
Comment (webberwang) [inline .circleci/config.yml:331]: We have `AWS_REGION`

**Review by webberwang — CHANGES_REQUESTED** [review]

**Pull request merged**